pkg/finddrive/finddrive.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package finddrive provides functionality to find an NVMe block device associated with
+// a particular physical slot on the machine, based on information in the SMBIOS table.
+package finddrive
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/smbios"
+)
+
+const (
+	// M2MKeySlotType is the SMBIOS slot type code for M.2 M-key
+	M2MKeySlotType = 0x17
+)
+
+// The path /sys/block/nvme0n1 will be a symlink to
+// /sys/devices/pci<domain:bus>/<slot BDF>/<drive BDF>/nvme/nvme0/nvme0n1
+func findBlockDevFromSmbios(sysPath string, s smbios.SystemSlots) ([]string, error) {
+	dev := (s.DeviceFunctionNumber & 0b11111000) >> 3
```

Comment:
add a comment for distribution of dev and fn bits in the DeviceFunctionNumber ?

Reply:
Added a comment.